From 4a8dc481c0f6fc6984ee784f7c2347ef33255a4e Mon Sep 17 00:00:00 2001
From: yxj-github-437 <2457369732@qq.com>
Date: Wed, 25 Mar 2026 02:29:14 +0800
Subject: [PATCH] use android liblog.so

---
 src/connection.c   | 12 +++++++++++-
 src/wayland-util.c | 11 +++++++++++
 2 files changed, 22 insertions(+), 1 deletion(-)

diff --git a/src/connection.c b/src/connection.c
index 2d1e8d1..dc26b58 100644
--- a/src/connection.c
+++ b/src/connection.c
@@ -42,6 +42,11 @@
 #include <time.h>
 #include <ffi.h>
 
+#ifdef __ANDROID__
+# define LOG_TAG "wayland"
+# include <android/log.h>
+#endif
+
 #include "wayland-util.h"
 #include "wayland-private.h"
 #include "wayland-os.h"
@@ -1660,7 +1665,12 @@ wl_closure_print(struct wl_closure *closure, struct wl_object *target,
 	fprintf(f, ")%s\n", color ? WL_DEBUG_COLOR_RESET : "");
 
 	if (fclose(f) == 0) {
-		fprintf(stderr, "%s", buffer);
+#ifdef __ANDROID__
+		if (!isatty(STDERR_FILENO)) {
+			__android_log_write(ANDROID_LOG_DEBUG, LOG_TAG, buffer);
+		} else
+#endif
+		fwrite(buffer, buffer_length, 1, stderr);
 		free(buffer);
 	}
 }
diff --git a/src/wayland-util.c b/src/wayland-util.c
index f551867..c17042f 100644
--- a/src/wayland-util.c
+++ b/src/wayland-util.c
@@ -31,6 +31,12 @@
 #include <string.h>
 #include <stdarg.h>
 
+#ifdef __ANDROID__
+# include <unistd.h>
+# include <android/log.h>
+# define LOG_TAG "wayland"
+#endif
+
 #include "wayland-util.h"
 #include "wayland-private.h"
 
@@ -452,6 +458,11 @@ wl_map_for_each(struct wl_map *map, wl_iterator_func_t func, void *data)
 static void
 wl_log_stderr_handler(const char *fmt, va_list arg)
 {
+#ifdef __ANDROID__
+	if (!isatty(STDERR_FILENO))
+		__android_log_vprint(ANDROID_LOG_ERROR, LOG_TAG, fmt, arg);
+	else
+#endif
 	vfprintf(stderr, fmt, arg);
 }
 
-- 
2.53.0.windows.2

