From 15ac6c4e6e50bcc79cb9b788d85a41a293fe0ecd Mon Sep 17 00:00:00 2001
From: yxj-github-437 <2457369732@qq.com>
Date: Fri, 14 Mar 2025 00:24:57 +0800
Subject: [PATCH] add undefined symbols before android24

---
 src/connection.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/connection.c b/src/connection.c
index 50d5fea..9bbccdb 100644
--- a/src/connection.c
+++ b/src/connection.c
@@ -50,6 +50,10 @@
 #include "wayland-private.h"
 #include "wayland-os.h"
 
+#if defined(__ANDROID__) && __ANDROID_API__ < 24
+FILE* open_memstream(char**, size_t*);
+#endif
+
 static inline uint32_t
 div_roundup(uint32_t n, size_t a)
 {
-- 
2.48.1.windows.1

